[timetraveller-123] iP

timetraveller-123 opened the pull request:
## **DukePro**
>"Your mind is for having ideas not holding them" - David Allen([source](https://dansilvestre.com/productivity-quotes/)]

DukePro frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ SUPER FAST to use

All you need to do is,
1) download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1) double-click it.
1) add your tasks.
1) let it manage your tasks for you 😉

And it is **FREE**!

## Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

##
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```Java
public static void main(String[] args) {

        String filePath = "./data/Duke.txt";
        Duke duke = new Duke(filePath);
        duke.run();

}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

timetraveller-123 committed:
* Rename bot to Vishnu
* Make bot greet and exit

timetraveller-123 committed:
* Make bot echo user input
* Make bot terminate with "bye" command

timetraveller-123 committed:
* Make bot add tasks to list
* Make bot list out tasks with "list" command

timetraveller-123 committed:
Add Task and Task List class
Make Bot mark and unmark tasks

timetraveller-123 committed:
Add Parser, different type of Task and Instruction class
Make bot accept different type of tasks

timetraveller-123 committed:
Add input and expected output

timetraveller-123 committed:
Add DukeException class
Make bot handle exception
Update Expected output

timetraveller-123 committed:
Add delete
Update input and expected output

timetraveller-123 committed:
Fix style mistakes

timetraveller-123 committed:
Add enum to represent Instuctions

timetraveller-123 committed:
Add storage to load and store data

timetraveller-123 committed:
Merge branch 'branch-Level-7'

timetraveller-123 committed:
Add LocalDateTime in Parser

timetraveller-123 committed:
Add LocalDateTime to tasks

timetraveller-123 committed:
Merge branch 'branch-Level-8'

timetraveller-123 committed:
Add Ui class

timetraveller-123 committed:
Merge master with branch-A-MoreOOP

timetraveller-123 committed:
Add packages

timetraveller-123 committed:
Merge branch 'branch-A-Packages'

timetraveller-123 committed:
Merge branch 'add-gradle-support'

timetraveller-123 committed:
Set up gradle

timetraveller-123 committed:
Merge branch 'branch-A-Gradle'

timetraveller-123 committed:
Add JUnit test

timetraveller-123 committed:
Merge branch 'branch-A-JUnit'

timetraveller-123 committed:
Update data path

timetraveller-123 committed:
Update text-ui-test

timetraveller-123 committed:
Update runtest.sh

timetraveller-123 committed:
Add find instruction

timetraveller-123 committed:
Add JavaDoc

timetraveller-123 committed:
Fix coding standard

timetraveller-123 committed:
Merge branch 'branch-Level-9'

timetraveller-123 committed:
Merge branch 'branch-A-JavaDoc'

timetraveller-123 committed:
Merge branch 'branch-A-CodingStandard'

HEEaZ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,91 @@
+     *
+     */
+    public void run() {
+        this.ui.showWelcome();
```
Since the variable `ui` is not being overshadowed, perhaps we can omit `this.ui` in favour of just `ui`?

HEEaZ commented on src/main/java/storagestuff/Storage.java:
```diff
@@ -0,0 +1,125 @@
+    public Storage(String filePath) throws DukeException {
+        this.FILEPATH = filePath;
+        try {
+
```
maybe we can omit this blank newline?

HEEaZ commented on src/main/java/taskstuff/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+            throw new DukeException("The index is not a valid index. Try again.");
+        }
+        tasks.get(index - 1).setAsNotDone();
+
```
Maybe we can omit the blank newline?

HEEaZ commented on src/main/java/userstuff/Ui.java:
```diff
@@ -0,0 +1,64 @@
+     */
+    public void showBye() {
+        System.out.println("\tBye. Hope to see you again soon!");
+        this.showLine();
```
since `showLine` is not being overshadowed, perhaps we can omit replace `this.showLine()` with just `showLine()`?

HEEaZ commented on src/main/java/taskstuff/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+    public void addTask(Task task) {
+        tasks.add(task);
+
+
```
Perhaps we can omit these blank newlines?

HEEaZ reviewed (commented):
Excellent code quality and style, you really adhered to the style guide well!

There's just a few minor space issues here and there, which probably can be fixed easily by a linter/ automatic code formatter :)

timetraveller-123 committed:
Add checkstyle and use follow checkstyle

timetraveller-123 committed:
Merge branch 'branch-A-CheckStyle'

timetraveller-123 committed:
Add gui

timetraveller-123 committed:
Merge branch 'branch-Level-10'

timetraveller-123 committed:
Add Varags

InstructionEnum: add other names for instructions

Each instruction is only referred to by only name.

More names for instructions is more user-friendly because users cannot be
expected to remember the exact instruction name.

Let's
* update find instruction to also match search
* update mark instruction to also match check and similary for unmark
* update delete instruction to also match remove

timetraveller-123 committed:
Merge branch 'branch-A-Varargs'

timetraveller-123 committed:
Refactor code to follow code quality standard

Parser: extract methods for parsing different instructions
  out of the main parse method

Method is very long and parse and parseData have duplicated code.

Let's,
* Extract parseDate and parseInteger method out
* Extract out parse methods for the different instructions

timetraveller-123 committed:
Fix method names in Parser

timetraveller-123 committed:
Fix parseDelete and parseUnmark

timetraveller-123 committed:
Add assertions

timetraveller-123 committed:
Add streams to remove some loops

timetraveller-123 committed:
Merge pull request #1 from timetraveller-123/branch-A-CodeQuality

Merge branch-A-CodeQuality

timetraveller-123 committed:
Merge branch 'master' into branch-A-Assertions

timetraveller-123 committed:
Merge branch 'master' into branch-A-Streams

timetraveller-123 committed:
Merge pull request #2 from timetraveller-123/branch-A-Assertions

Merge branch-A-Assertions

timetraveller-123 committed:
Merge branch 'master' into branch-A-Streams

timetraveller-123 committed:
Merge pull request #3 from timetraveller-123/branch-A-Streams

Merge branch-A-Streams

timetraveller-123 committed:
Make finding instuction case-insensitive

Find Instruction: make matching case-insensitive

Find Instruction is case-sensitive.

A case-insensitive instruction is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's,
* update the search algorithm to use case-insensitive matching

timetraveller-123 committed:
Add undo

timetraveller-123 committed:
Merge branch 'branch-BCD-Extension'

timetraveller-123 committed:
Make display not truncate

timetraveller-123 committed:
Make bot say exit message

timetraveller-123 committed:
Add title to window

timetraveller-123 committed:
Fix deadline parse

timetraveller-123 committed:
Add README.md

timetraveller-123 committed:
Fix image path

timetraveller-123 committed:
Add welcome pic

timetraveller-123 committed:
Fix welcome pic

timetraveller-123 committed:
Fix README.md

timetraveller-123 committed:
Fix some javadocs

timetraveller-123 committed:
Add link to jar in user guide

timetraveller-123 committed:
Fix readme

timetraveller-123 committed:
Fix test and fix style

tjingsheng commented on src/main/java/instructionstuff/Instruction.java:
```diff
@@ -0,0 +1,288 @@
+package instructionstuff;
+
```
I would think it's better to have the package without the word "stuff" as it is redundant

tjingsheng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,305 @@
+ * or not.
+ */
+public class Parser {
+
```
I like your parser class. It's quite different from how I implemented mine.

tjingsheng commented on src/main/java/instructionstuff/Instruction.java:
```diff
@@ -0,0 +1,288 @@
+        @Override
+        public void execute(Storage storage, TaskList taskList, MainWindow mainWindow) throws DukeException {
+            taskList.unmarkTask(this.index);
+            assert index >= 1 && index <= taskList.getSize();
```
I am not sure why the assertion is done before the unmarking, it's also quite interesting that the same conditions are checked within execute, and and exception is thrown.

if the index is indeed invalid, an exception is thrown in taskList.unmarkTask. Then is the assert still necessary? I am not too sure

tjingsheng commented on src/main/java/taskstuff/Event.java:
```diff
@@ -0,0 +1,45 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + startTime.format(Parser.OUTPUT_FORMAT)
+                + " to: " + endTime.format(Parser.OUTPUT_FORMAT) + ")";
```
is there a way to make this more readable?

tjingsheng commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,72 @@
+
+    @Test
+    public void parse_byeInstruction_success() {
+        try {
```
Does these test names follow method conventions?

tjingsheng reviewed (commented):
Well done!

timetraveller-123 committed:
Fix corrupted data loading
